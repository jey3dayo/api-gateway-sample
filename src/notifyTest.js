import runWarm from './utils/runWarm';
import { successResponse } from './utils/lambdaResponse';

const AWS = require('aws-sdk');

const lambda = new AWS.Lambda();

const notifyTest = async event => {
  const query = event?.queryStringParameters ?? null;

  const payload = {
    message: query,
  };

  const params = {
    FunctionName: 'caproni-sample-development-hello',
    InvocationType: 'RequestResponse',
    Payload: JSON.stringify(payload),
  };

  let callLambda = null;
  try {
    // Lambda関数呼び出し
    callLambda = await lambda.invoke(params).promise();

    console.log(callLambda);
  } catch (e) {
    console.log('[ERROR]hello/callLambda', e);
  }

  return successResponse({ query, callLambda });
};

export default runWarm(notifyTest);
