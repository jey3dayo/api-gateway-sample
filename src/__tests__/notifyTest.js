import notifyTest from '../notifyTest';

describe('hello', () => {
  it('executes as expected', async () => {
    const response = await notifyTest({});
    expect(response).toMatchSnapshot();
  });

  it('executes as expected include params', async () => {
    const event = {
      queryStringParameters: {
        test: 'example'
      }
    }
    const response = await notifyTest(event);
    expect(response).toMatchSnapshot();
  });
});
