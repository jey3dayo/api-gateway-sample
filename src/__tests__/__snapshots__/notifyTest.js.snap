// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello executes as expected 1`] = `
Object {
  "body": "{\\"message\\":{\\"NODE_ENV\\":\\"development\\"}}",
  "statusCode": 200,
}
`;

exports[`hello executes as expected include params 1`] = `
Object {
  "body": "{\\"message\\":{\\"NODE_ENV\\":\\"development\\",\\"query\\":{\\"test\\":\\"example\\"}}}",
  "statusCode": 200,
}
`;
